code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -102,17 +135,29 @@
 		if(mode == 2)
 			point_at(target)
 		to_chat(usr, "<span class='notice'>You activate the pinpointer.</span>")
+		START_PROCESSING(SSfastprocess, src)
 	else
 		active = 0
 		icon_state = icon_off
 		to_chat(usr, "<span class='notice'>You deactivate the pinpointer.</span>")
+		STOP_PROCESSING(SSfastprocess, src)
+
+/obj/item/pinpointer/advpinpointer/process()
+	if(active == 1)
```

Comment:
Can you make it so you extract the active code into a proc? Both for this pinpointer instance and the base one.
This way you can just have one process at the base which does something like this:
```
if(active)
	do_active()
else
	do_inactive() // if needed
```

Reply:
changed to stuff discussed over discord